### PR TITLE
Socket support for manage unit and dropin

### LINKED ISSUE #348: Feature Request for socket unit files
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 6.x
- Ruby: 3.x
- Distribution: debian
- Module version: 5.0.0

Im missing a possibility to create socket unit files in the systemd module. As an example:

/etc/systemd/system/pdnsadmin.socket
```ini
[Unit]
Description=PowerDNS-Admin socket

[Socket]
ListenStream=/run/pdnsadmin/socket

[Install]
WantedBy=sockets.target
```

Would be very helpfull if the module would be extended with this.

### PR DESCRIPTION
**Pull request by traylenator**
#### Pull Request (PR) description

A socket file can now be created:

```puppet
systemd::manage_unit{'arc.socket':
  ensure => present,
  unit_entry => {
    'Description' => 'My great socket',
  },
  socket_entry => {
    'Type'         => 'simple',
    'ExecStart'    => '/usr/sbin/arcd /usr/libexec/arcd/arcd.pl',
    'StandardInput => 'socket',

  }
```

All socket directives from 253.

* https://www.freedesktop.org/software/systemd/man/systemd.socket.html

#### This Pull Request (PR) fixes the following issues
Fixes #348

### CODE REVIEW

**Review by smortex — APPROVED** [review]
Looks great!

**Comment by mplaats** [comment]
Works perfect! For reference the hiera code used to test the new feature.

```
systemd::manage_unit:
  'pdnsadmin.socket':
    ensure: present
    unit_entry:
      'Description': 'PowerDNS-Admin socket'
    socket_entry:
      'ListenStream': '/run/pdnsadmin/socket'
    install_entry:
      'WantedBy': 'sockets.target'
```

File: manifests/manage_unit.pp [hunk 1]
```diff
@@ -21,18 +21,48 @@
...
+# @example Generate a socket and service (xinetd style)
+#  systemd::manage_unit {'arcd.socket':
+#    ensure        => 'present',
+#    unit_entry    => {
+#      'Description' => 'arcd.service',
+#    }
+#    socket_entry  => {
+#      'ListenStream' => 4241,
+#      'Accept'       => true,
+#      'BindIPv6Only' => 'both',
```
Comment (ekohl) [inline manifests/manage_unit.pp:45]: Late to the party, but I think this is missing a closing `}`.

Comment (ekohl) [reply manifests/manage_unit.pp:45]: https://github.com/voxpupuli/puppet-systemd/pull/354

**Pull request merged**